devtools/config-doc-maven-plugin/src/main/java/io/quarkus/maven/config/doc/generator/MarkdownFormatter.java

```diff
@@ -17,17 +16,13 @@ protected JavadocFormat javadocFormat() {
         return JavadocFormat.MARKDOWN;
     }
 
-    @Override
-    public String formatSectionTitle(ConfigSection configSection) {
-        // markdown only has 6 heading levels
-        int headingLevel = Math.min(6, 2 + configSection.getLevel());
-        return "#".repeat(headingLevel) + " " + super.formatSectionTitle(configSection);
-    }
-
     @Override
     public String escapeCellContent(String value) {
         String cellContent = super.escapeCellContent(value);
-        return cellContent == null ? null : cellContent.replace("\n\n", "<br><br>").replace("\n", " ");
+        return cellContent == null ? null
+                : cellContent
+                        .replaceAll("(\r\n)+", " ")
+                        .replaceAll("\n+", " ");
```

Comment:
Are you sure about that? I would expect us to allow having several paragraphs?

What are you trying to solve exactly?

Reply:
I wanted to replicate the behavior of the asciidoc formatter ([source](https://github.com/Athou/commafeed/blob/master/commafeed-server/src/main/java/com/commafeed/CommaFeedConfiguration.java#L32-L36), [output](https://github.com/Athou/commafeed/blob/master/commafeed-server/doc/commafeed.adoc)) for consistency. I agree, if there are paragraphs in the javadoc it would be best to keep those paragraphs in the documentation. I can remove the method override altogether.

Reply:
Hmmm, I think it's more a bug in the Asciidoc output. No idea where it comes from though. If you're interested in having a look, patch is welcome :).

I think you can drop the method and we need to do better on the Asciidoc side.

Reply:
I dropped the method yesterday :)

As for the issue with the asciidoc paragraphs, I tracked the issue to the `TEXT_NODE` case of `io.quarkus.annotation.processor.documentation.config.formatter.JavadocToAsciidocTransformer#appendHtml`. Looks like the `text()` method of JSoup removes line breaks, `getWholeText()` should be called to preserve them, but it breaks some tests and I'm not familiar enough with asciidoc to fix this.